### PR TITLE
Gather file descriptors as process metric

### LINKED ISSUE #729: Gather number of open FDs & max number as a metric

### PR DESCRIPTION
**Pull request by paulbellamy**
Fixes #729 

Any other bits we should to gather? I'm not super-convinced it's actually all that useful?

@2opremio, wdyt?

See "Open Files" sparkline below.
screen shot 2016-02-12 at 17 03 39

### CODE REVIEW

**Comment by 2opremio** [comment]
I think it's a  very useful metric, but I am not convinced it should have
such protagonism, or whether it's useful to show it it's history in a
sparkline.

**Comment by tomwilkie** [comment]
Yes it would be good to have this hidden behind a show more.

**Comment by paulbellamy** [comment]
Hiding a single element seems needless.

**Comment by davkal** [comment]
@paulbellamy From my own debugging experience, open files gets you straight into root cause analysis, whereas cpu and memory are quite vague. so I love its prominence there (until a better metric comes along).

**Comment by davkal** [comment]
Tried the branch, JS part looks good too.

File: probe/process/walker_linux.go [hunk 1]
```diff
@@ -83,6 +88,11 @@ func (w *walker) Walk(f func(Process, Process)) error {
 			continue
 		}
 
+		fileDescriptors, err := readFileDescriptors(path.Join(w.procRoot, filename, "fd"))
```
Comment (2opremio) [inline probe/process/walker_linux.go]: I would inline the function call. It's just two lines and the parameter is not obvious.

File: probe/process/walker.go [hunk 2]
```diff
@@ -4,12 +4,13 @@ import "sync"
...
-	Threads   int
-	Jiffies   uint64
-	RSSBytes  uint64
+	PID, PPID       int
+	Name            string
+	Cmdline         string
+	Threads         int
+	Jiffies         uint64
+	RSSBytes        uint64
+	FileDescriptors int
```
Comment (2opremio) [inline probe/process/walker.go]: Maybe rename it to `OpenFilesCount` or similar? It's more accurate.

**Comment by 2opremio** [comment]
minor-comments aside, LGTM

**Pull request merged**